app/models/Tokens.py

```diff
@@ -0,0 +1,191 @@
+import os
+from datetime import datetime, timedelta
+
+import bcrypt
+import jwt
+import redis
+from dotenv import load_dotenv
+from fastapi import HTTPException
+from passlib.context import CryptContext
+
+import enum
+from typing import Final
+
+from app.utils.databases.redis import Redis
+from app.utils.http_errors import CommonErrorMessages, ClassicExceptions
+
+pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
+
+class TokenTypes(enum.Enum):
+    """
+    This enumeration provides the different token types available.
+    These types have different secrets and expiration time.
+    """
+    AUTH_TOKEN:     Final[str] = 'AUTH_TOKEN'
+    REFRESH_TOKEN:  Final[str] = 'REFRESH_TOKEN'
+
+class TokenAttributes:
+
+    token_type:  TokenTypes
+    expire_time: str
+    secret:      str
+
+    def __init__(self, token_type: TokenTypes):
+        self.token_type = token_type                                # <- Type of the token. Used to get specific env vars.
+
+        load_dotenv()
+        self.algorithm  = os.environ.get("JWT_ALGORITHM", "HS256")  # <- Algorithm used for encoding the tokens
+        self.expire_time = os.environ.get(token_type.value + "_EXPIRE", None)
+        if self.expire_time is None:
+            raise Exception("No expire time was provided inside the .env file !")
+
+        self.secret = os.environ.get("JWT_SECRET_" + token_type.value, None)
+        if self.expire_time is None:
+            raise Exception("No secrets were provided inside the .env file !")
+
+class AvailableTokenAttributes(enum.Enum):
+    """
+    This enumeration lists the available token attributes.
+    It initializes the values of these attributes with data provided inside the .env file.
+    """
+    AUTH_TOKEN:    Final[TokenTypes] = TokenAttributes(TokenTypes.AUTH_TOKEN)
+    REFRESH_TOKEN: Final[TokenTypes] = TokenAttributes(TokenTypes.REFRESH_TOKEN)
+
+
+# -------- Classic models -------- #
+class Token:
+    value: str | None = None
+    attributes: TokenAttributes
+
+    def __init__(self, attributes: TokenAttributes, value: str | None = None):
+        self.attributes = attributes
+        self.value      = value
+
+    def generate(self, user_id: int):
+        """
+        This method generates a single token.
+        It uses the attributes specified to generate the correct token using the correct private key.
+        :return: A Json Web Token.
+        """
+        # Generating data for the token
+        creation_date = datetime.now()
+        expire_date = datetime.now() + timedelta(minutes=float(self.attributes.expire_time))
+        jwt_data = {
+            "user_id": user_id,
+            "salt": str(bcrypt.gensalt()),
+            "iat": creation_date,
+            "exp": expire_date
+        }
+        self.value = jwt.encode(jwt_data, str(self.attributes.secret), self.attributes.algorithm)
+
+    async def revoke(self, redis_db: redis.Redis = Redis.get_redis()):
```

Comment:
should be typed with Awaitable[\<ReturnType>] or Coroutine[\<YieldType>, \<SendType>, \<ReturnType>]